### PR TITLE
fix(sandpack-react): tweak imports to avoid circular deps

### PR DESCRIPTION
**Pull request by jeetiss**
reduce the number of circular dependencies

## What is the current behavior?

```
(!) Circular dependencies
src/index.ts -> src/components/index.ts -> src/components/FileExplorer/index.tsx -> src/components/common/index.ts -> src/components/common/ErrorOverlay.tsx -> src/index.ts
src/index.ts -> src/components/index.ts -> src/components/FileExplorer/index.tsx -> src/components/common/index.ts -> src/components/common/LoadingOverlay.tsx -> src/index.ts
src/index.ts -> src/components/index.ts -> src/components/FileExplorer/index.tsx -> src/components/common/index.ts -> src/components/common/LoadingOverlay.tsx -> src/hooks/usePreviewProgress.ts -> src/index.ts
...and 4 more
```

## What is the new behavior?

```
(!) Circular dependency
src/components/FileExplorer/ModuleList.tsx -> src/components/FileExplorer/Directory.tsx -> src/components/FileExplorer/ModuleList.tsx
```

## What steps did you take to test this?

- build the package

### CODE REVIEW

File: sandpack-react/src/hooks/usePreviewProgress.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import type { WorkerStatusUpdate } from "@codesandbox/nodebox";
```
Comment (danilowoz) [inline sandpack-react/src/hooks/usePreviewProgress.ts:1]: Would you mind renaming this hook to `useSandpackPreviewProgress` and exporting it on `hooks/index.ts`?

**Review by danilowoz — APPROVED** [review]
Thank you!

File: sandpack-react/src/hooks/usePreviewProgress.ts (hunk 1 above)
Comment (jeetiss) [reply sandpack-react/src/hooks/usePreviewProgress.ts:1]: of course

**Review by danilowoz — APPROVED** [review]
Thank you!

**Pull request merged**